### ISSUE TITLE
[Usage]: How to get original total hidden_states

### ISSUE THREAD

**Issue by world1256** [vllm-project/vllm#14607]
### Proposal to improve performance

I want to get the original total hidden_states like transformers output, here is my transformer example:

`from transformers import AutoModel

base_model_path = "../Llama-3.2-3B-Instruct"
base_model = AutoModel.from_pretrained(base_model_path,torch_dtype=torch.float32)
base_model.to("cuda")

inputs = tokenizer("what's the weather today",return_tensors="pt")
cuda_inputs ={}
for key in inputs:
    cuda_inputs[key] = inputs[key].to(peft_model.device)

with torch.no_grad():
    outputs = base_model (**cuda_inputs)
    print(outputs[0])`

The output likes blow:

`tensor([[[ 0.0127, -0.4116, -0.2221,  ...,  0.4073,  0.3212,  0.2383],
         [-1.7105, -1.9136,  2.1987,  ..., -1.1063, -1.1604,  1.9475],
         [-2.3868, -0.6762,  0.6541,  ..., -1.8436, -2.8037, -0.2768],
         [-1.2720, -1.5623,  1.6725,  ..., -0.5808, -2.6175,  1.5599],
         [-1.3572, -3.6725,  2.8592,  ..., -1.5065, -1.3943,  1.0695],
         [ 0.0378, -1.8762,  3.3803,  ..., -2.6625, -0.3675,  3.0536]]],
       device='cuda:0')`
The shape of the  output is [1,6,3072]. Because the input seq lenth is 6, and the batch_size is 1.

**How can I get this result in vllm?**

I have tried the embed and encode method in vllm. But the output shape is [3072]. And is different from the output shape in transformer. And the value is not same as the last index of the original outputs.

`'''the last index of the original outputs from transformer'''
[ 0.0378, -1.8762,  3.3803,  ..., -2.6625, -0.3675,  3.0536]

'''output from vllm encode'''
tensor([ 0.0003, -0.0215,  0.0356,  ..., -0.0302, -0.0004,  0.0374])`

And the vllm code is like blow:
`from vllm import LLM, SamplingParams
from vllm.lora.request import LoRARequest
import transformers

base_model_path = "../Llama-3.2-3B-Instruct"
llm = LLM(model=base_model_path ,max_model_len=2048,task="embed",dtype='float32')
encode_outputs = llm.encode(prompts)
encode_outputs[0].outputs.data`

### Your current environment (if you think it is necessary)

```text
The output of `python collect_env.py`
```

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by DarkLight1337** [comment]
As noted [here](https://docs.vllm.ai/en/latest/models/pooling_models.html), you can use `--task reward` so that `encode` returns the raw hidden states.

**Comment by world1256** [comment]
> As noted [here](https://docs.vllm.ai/en/latest/models/pooling_models.html), you can use `-task="reward"` so that `encode` returns the raw hidden states.

It's worked for me,  thank you very much.

**Comment by world1256** [comment]
Just use -task="reward" so that encode returns the raw hidden states.
